src/_DataStructures_/Trees/Trie/get-unique-words/get-unique-words.test.js

```diff
@@ -11,6 +11,52 @@ describe('Data Structure : Trie : Get unique words', () => {
     const result = getUniqueWords(trie.root);
 
     const expected = ['apple', 'ball', 'bed', 'java', 'javascript'];
-    expect(expected).toEqual(result);
+    expect(result).toEqual(expected);
+  });
+
+  it('removes duplicates', () => {
+    const words = ['bed', 'bed', 'bed'];
+    const trie = new Trie();
+
+    words.forEach(word => trie.insert(word));
+
+    const result = getUniqueWords(trie.root);
+    expect(result.length).toBe(1);
+  });
+
+  it('passing an empty array of words returns an empty array', () => {
+    const words = [];
+    const trie = new Trie();
+
+    words.forEach(word => trie.insert(word));
+
+    const result = getUniqueWords(trie.root);
+    expect(result).toEqual([]);
+  });
+
+  it('passing an empty Trie will throw an error ', () => {
+    const trie = new Trie();
+    expect(() => {
+      getUniqueWords(trie);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+  it('passing an empty array will throw an error ', () => {
+    expect(() => {
+      getUniqueWords([]);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+  it('passing null will throw an error ', () => {
+    expect(() => {
+      getUniqueWords([]);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+
+  it('passing an array not in a Trie will throw an error ', () => {
+    expect(() => {
+      getUniqueWords(['bed', 'ball', 'apple']);
+    }).toThrow('Cannot read property \'0\' of undefined');
```

Comment:
Thanks for highlighting, this case was something not handled by me in the implementation. Let me add an error message or a better approach is to return an empty array. 

Accordingly, you will have to take a pull of master and run the tests again, these selected tests will fail and you will just have to update the test cases with the result. 

Thank you again.

Reply:
Code will throw **"Invalid argument: Root of Trie is required"**, please make changes accordingly. Do not forget to take the pull of master: **git pull origin master**

Reply:
I've updated this branch and amend tests to expect above error